ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/AutoCreateIcebergTagAction.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.server.optimizing.maintainer;
+
+import com.netease.arctic.server.table.TagConfiguration;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.Table;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+
+/** Action to auto create tag for Iceberg Table. */
+public class AutoCreateIcebergTagAction {
+  private static final Logger LOG = LoggerFactory.getLogger(AutoCreateIcebergTagAction.class);
+
+  private final Table table;
+  private final TagConfiguration tagConfig;
+  private final LocalDateTime now;
+
+  public AutoCreateIcebergTagAction(Table table, TagConfiguration tagConfig, LocalDateTime now) {
+    this.table = table;
+    this.tagConfig = tagConfig;
+    this.now = now;
+  }
+
+  public void execute() {
+    if (!tagConfig.isAutoCreateTag()) {
+      return;
+    }
+    LOG.info("Start checking the automatic creation of tags for {}", table.name());
+    if (tagExist()) {
+      LOG.debug("{} finds expect tag, skip", table.name());
```

Comment:
nit: The log message here confuses me a bit. It's not that the table found the tag, but rather that the tag was found on the table.